### ISSUE TITLE
Export libc++ symbols for native modules

### ISSUE THREAD

**Issue by implausible** [electron/electron#27269]
### Preflight Checklist

* [X] I have read the [Contributing Guidelines](https://github.com/electron/electron/blob/master/CONTRIBUTING.md) for this project.
* [X] I agree to follow the [Code of Conduct](https://github.com/electron/electron/blob/master/CODE_OF_CONDUCT.md) that this project adheres to.
* [X] I have searched the issue tracker for an issue that matches the one I want to file, without success.

### Issue Details

* **Electron Version:**
  * 10.1.4 and above
* **Operating System:**
  * Windows, primarily
* **Last Known Working Electron version:**
API is only exposed in 10.1.4 and above.

When compiling a native node module using node-gyp on Windows, the node module cannot be compiled due to missing symbols. I suspect because Electron uses a custom built libc++ for Windows, it will no longer be possible to compile native node modules for windows except in cases that do not allow for use of the standard library.

The particular API I need to use is `AsyncCleanupHookHandle node::AddEnvironmentCleanupHook(v8::Isolate **isolate, AsyncCleanupHook fun, void *args);`. This is critical for my library NodeGit when the context is being shut down.

Basically, because AsyncCleanupHookHandle boils down to `std::unique_ptr<ACHHandle, DeleteACHHandle>`, it is not possible for this API to be leveraged by a native node module without that native node module also being compiled for the same standard library.

Unfortunately due to the complexity of the library, refactoring to N-API is not likely to be a great solution for me. We do a lot of lower level work specifically with v8, garbage collection, and callbacks that are going to require a lot more than what n-api currently provides us.

I suspect what needs to happen is the headers for libc++ that are compiled on Windows need to be exported in the build process so that interested parties like NodeGit can disable compilation for the stdlib and leverage the libc++ headers from Electron instead.

### Expected Behavior
Native node module compiles and works.

### Actual Behavior
Native node module fails to link, because of a missing symbol.

### To Reproduce
This repository reproduces the issue at hand:
https://github.com/implausible/electron-abi-issue-demo

Check the README to see repro steps.

### Additional Information
I believe this issue is related https://github.com/electron/electron/issues/22799
I've also opened a corresponding issue on nodejs, specifically because this is so not compatible I was seeking perhaps a different implementation of the async cleanup hook... https://github.com/nodejs/node/issues/36349#issuecomment-736887148

I'm also pinging @deepak1556 @codebytere to get some eyes on this. This is kind of an existential threat for NodeGit and the product that I develop GitKraken. Especially with the libuv big sur slowdowns (we're worried that the libuv backport won't make it into 10 and thus it's _super pressing_ for us to get onto Electron 11 asap, which we're blocked by this issue now).

I'm working through a lot of what I think is ground work for me (getting onto clang-cl, building electron, learning what flags I can pass to clang-cl to make this compile), but what we're missing is the headers and defines that are necessary to build against Electron now.

**Comment by implausible** [comment]
It also looks like we're missing the libc++.lib in general. I'm learning the process for building Electron and wow I've got a long way to go here. I've done a lot of local hacking to get to a state where my nodegit build just wants libc++.lib (assuming I don't also have to throw the custom windows sdk at it), we'll find out after I get through building Electron today to see if the artifact _ever_ pops out.

@jkleinsc I am not sure if I would call this an enhancement or broken... I think I labeled the issue with what my suggestion for fixing it is, but as it stands today, NAN based native node modules that rely on async clean up behavior to qualify for context-awareness cannot be compiled for Electron.
